src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -1082,12 +1082,28 @@ extern "C" int32_t SystemNative_CopyFile(intptr_t sourceFd, intptr_t destination
     while (CheckInterrupted(ret = fstat_(inFd, &sourceStat)));
     if (ret == 0)
     {
+#if HAVE_FUTIMES
         struct timeval origTimes[2];
-        origTimes[0].tv_sec = sourceStat.st_atime;
         origTimes[0].tv_usec = 0;
-        origTimes[1].tv_sec = sourceStat.st_mtime;
         origTimes[1].tv_usec = 0;
+#elif HAVE_FUTIMENS
```

Comment:
Since futimens has nanosecond precision and futimes "only" microsecond, I wonder if we should switch the precedence, and prefer futimens if it exists on the target platform?